goldstone/core/tasks.py

```diff
@@ -150,3 +151,49 @@ def log_event_search():
 
     cadf = CADFEventDocType(event=e)
     return cadf.save()
+
+
+@celery_app.task()
+def check_for_pending_alerts():
+    """
+     Run an AlertSearch query to check for any pending alerts to be fired.
+    """
+
+    saved_alerts = AlertSearch.objects.filter()
+
+    for obj in saved_alerts:
+        # execute the search, and assuming no error, update the last_ times
+        s, start, end = obj.search_recent()
+        response = s.execute()
+        if response.hits.total > 0:
+            # We have a non-zero match for pending alerts
+            # Go ahead and generate an instance of the alert object here.
+            # We can directly call the producer class to send an email
+
+            now = arrow.utcnow().datetime
+
+            msg_dict = obj.build_alert_message_template(hits=
+                                                        response.hits.total)
+
+            # For this scheduled celery task, pick up the message template
+            # from the query object and pass it along. For all other
+            # cases, user is allowed to send custom msg_title and msg_body
+            alert_obj = Alert(name='Scheduled Alert loop :' + str(now),
+                              query=obj, msg_title=msg_dict['title'],
+                              msg_body=msg_dict['body'])
+
+            # Currently we only have a producer interface for emails.
+            # For the default celery task, goldstone-bot@solinea.com acts as
+            # both the sender and receiver.
+            producer_obj = EmailProducer('goldstone-bot@solinea.com', 'goldstone-bot@solinea.com')
```

Comment:
this should simply be a loop of all producers with an FK to obj.  Don't create the producer here.